adding analytics for tsp

therocket-gr opened the pull request:
Related to #

3113-gh-tumblr/a8c-dsp

## Proposed Changes

adding analytics events for tsp related events


## Why are these changes being made?

we want to change the wording on this link

<img width="315" alt="Screenshot 2025-02-17 at 12 50 27 PM" src="https://github.com/user-attachments/assets/ab908ca5-d47e-495c-a135-5ae7b7849b8f" />

we also want to add analytics events when a click is registered and when the user scrolls to get the tsp data into view



## Testing Instructions

1. make your window short so that you can simulate scroll into view (you will need an active promoted post campaign with tsp - if you can get an old one that has some stats for the normal campaign it is better - so that the stats section is populated)

open network inspection and scroll the tsp section into view - check the network to see the event firing


<img width="375" alt="Screenshot 2025-02-17 at 12 51 48 PM" src="https://github.com/user-attachments/assets/5b3a6e00-5f5d-47ca-862e-6f3360911ed7" />


click on the above link
<img width="382" alt="Screenshot 2025-02-17 at 12 52 01 PM" src="https://github.com/user-attachments/assets/4c4667e1-8f7e-4d47-ab16-0e67d9dd39d4" />

do a CR

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

therocket-gr committed:
adding analytics for tsp

therocket-gr committed:
minor fix in typo

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -608,6 +609,28 @@ export default function CampaignItemDetails( props: Props ) {
 
+	const tspTargetRef = useRef< HTMLDivElement | null >( null );
+
+	useEffect( () => {
```
We should probably include a condition to check if the stats endpoint finished loading. 
In my case, I loaded the page and triggered the event, and after a couple of seconds, the page accommodated to the graph, and the section was moved down.

Not a blocker. It will be hard to adapt to all the cases (e.g., campaigns in moderation, canceled, etc.)

therocket-gr replied:
added a condition 
```
if ( ! isLoading && ( ! hasStats || ! campaignsStatsIsLoading) ) {
```

it will check if isLoading is complete. in case we have stats it will wait for campaignsStatsIsLoading to be completed

sbarbosa reviewed (approved):
Code looks good and works as described. 

I added a comment, but its not a blocker.

One thing to consider is that it would be nice to also track clicks on the Learn More link (for the new banner), and in the Show More link in this page.

timur987 reviewed (approved):
Thanks @therocket-gr!

The code LGTM and works as advertised (the link caption was changed, the on-scroll event was sent) 👍

therocket-gr commented:
> Code looks good and works as described.
> 
> I added a comment, but its not a blocker.
> 
> One thing to consider is that it would be nice to also track clicks on the Learn More link (for the new banner), and in the Show More link in this page.

i added another event in the show more (for the tsp replies) but I am not sure about the new banner.. i don't see a link in there
ss from figma looks like 
<img width="835" alt="Screenshot 2025-02-17 at 6 51 15 PM" src="https://github.com/user-attachments/assets/af19c583-e054-4e46-a65c-2980073d29c0" />

therocket-gr committed:
minor fix to add the eventlister after everything has loaded
- added a new event in show more click on the replies button

therocket-gr committed:
added a tracks event in tsp banner learn more button to track clicks

therocket-gr commented:
> @therocket-gr, that is a banner that we show on the campaign details page. Its not yet implemented. Barts left us a comment on the designs P2, but we missed it when we work on the banner and the campaign details page.
> 
> I already created the ticket for that, but we do have a banner on the dashboard. We could include it on that banner, and then copy the logic when we work on the banner in the campaign details page.
> 
> Link: https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/promote-post-i2/components/tsp-banner/index.tsx

added the tracks event here
https://github.com/Automattic/wp-calypso/pull/99861/commits/df2810226fdb6eb8a8e0d211b1a95ea7db9e7bea

@sbarbosa

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17259822

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @therocket-gr for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17259822).
